includes/components/class-geolocation.php

```diff
@@ -107,61 +110,63 @@ public function add_attributes( $attributes ) {
 			$radius *= 1.60934;
 		}
 
-		return array_merge(
-			$attributes,
-			[
-				'location'  => [
-					'editable'     => true,
-					'searchable'   => true,
-
-					'edit_field'   => [
-						'label'     => esc_html__( 'Location', 'hivepress-geolocation' ),
-						'type'      => 'location',
-						'countries' => $countries,
-						'required'  => true,
-						'_order'    => 35,
-					],
+		// Set fields.
+		$fields = [
+			'latitude'  => [
+				'editable'     => true,
+				'searchable'   => true,
 
-					'search_field' => [
-						'placeholder' => esc_html__( 'Location', 'hivepress-geolocation' ),
-						'type'        => 'location',
-						'countries'   => $countries,
-						'_order'      => 20,
-					],
+				'edit_field'   => [
+					'label' => esc_html__( 'Latitude', 'hivepress-geolocation' ),
+					'type' => 'latitude',
 				],
 
-				'latitude'  => [
-					'editable'     => true,
-					'searchable'   => true,
+				'search_field' => [
+					'type'   => 'latitude',
+					'radius' => $radius,
+				],
+			],
 
-					'edit_field'   => [
-						'label' => esc_html__( 'Latitude', 'hivepress-geolocation' ),
-						'type' => 'latitude',
-					],
+			'longitude' => [
+				'editable'     => true,
+				'searchable'   => true,
 
-					'search_field' => [
-						'type'   => 'latitude',
-						'radius' => $radius,
-					],
+				'edit_field'   => [
+					'label' => esc_html__( 'Longitude', 'hivepress-geolocation' ),
+					'type' => 'longitude',
 				],
 
-				'longitude' => [
-					'editable'     => true,
-					'searchable'   => true,
+				'search_field' => [
+					'type'    => 'longitude',
+					'radius'  => $radius,
+					'_parent' => 'latitude',
+				],
+			],
+		];
 
-					'edit_field'   => [
-						'label' => esc_html__( 'Longitude', 'hivepress-geolocation' ),
-						'type' => 'longitude',
-					],
+		if ( in_array( 'location', (array) get_option( 'hp_listing_search_fields' ) ) ) {
```

Comment:
The Location attribute shouldn't be removed completely, e.g. it should stay in the listing edit form, please remove it from the search form only. I assume this can be done if you set the "searchable" parameter depending on if the Location field is in Default Fields or not, this parameter defines the attribute visibility in the main search form.